api/tests/unit/environments/test_unit_environments_models.py

```diff
@@ -1252,16 +1252,26 @@ def test_environment_clone_from_non_versioned_environment_with_use_v2_feature_ve
     project: Project,
     environment: Environment,
     feature: Feature,
+    segment_featurestate: FeatureState,
     enable_v2_versioning_for_new_environments: typing.Callable[[], None],
 ) -> None:
     # Given
+    assert not environment.use_v2_feature_versioning
```

Comment:
Seems more resilient/correct:

```suggestion
    environment.use_v2_feature_versioning = False
    environment.save()
```

Reply:
Hmm... I'm not sure I agree with this. The assertion is there to ensure that there's nothing going awry with the fixture and I want to know about it if there is, and implementing the change suggested here would just mask it.